exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -386,6 +380,42 @@ def as_otlp_any_value(value)
           result
         end
 
+        # Returns the url used by the exporter based on the precedence defined in the specification:
+        # https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#endpoint-urls-for-otlphttp
+        #
+        # @param [String, nil] endpoint
+        # @return [URI::Generic]
+        def determine_url(endpoint)
+          if !endpoint.nil?
+            # Use endpoint provided via argument as-is
+            parse_url_or_raise(endpoint)
+          elsif !ENV['OTEL_EXPORTER_OTLP_TRACES_ENDPOINT'].nil?
+            # Use signal-specific endpoint as-is
+            parse_url_or_raise(ENV['OTEL_EXPORTER_OTLP_TRACES_ENDPOINT'])
+          elsif !ENV['OTEL_EXPORTER_OTLP_ENDPOINT'].nil?
+            # Append v1/traces only if the non-signal specific env var is set.
+            non_signal_specific_endpoint = ENV['OTEL_EXPORTER_OTLP_ENDPOINT']
+            # append '/' if not present
+            non_signal_specific_endpoint += '/' unless non_signal_specific_endpoint.end_with?('/')
+            begin
+              URI.join(non_signal_specific_endpoint, 'v1/traces')
+            rescue URI::InvalidURIError
+              raise ArgumentError, "invalid url for OTLP::Exporter #{non_signal_specific_endpoint}"
```

Comment:
I strongly prefer to `raise ArgumentError` from the public method rather than a private one. It is more informative for the caller. Since this case is so specific, I think it would also be useful to call out the environment variable, in case the user wonders where the value came from.

Reply:
I'm re-raising in the constructor now to hide the private method [f2748b7](https://github.com/open-telemetry/opentelemetry-ruby/pull/1506/commits/f2748b7f89d29c61f6d255d051ddcbee208692fc), couldn't pull everything in the constructor unfortunately as complexity ended up being too high and rubocop failed.